lib/spack/spack/solver/asp.py

```diff
@@ -3317,6 +3317,13 @@ def _is_reusable(spec: spack.spec.Spec, packages, local: bool) -> bool:
     if "dev_path" in spec.variants:
         return False
 
+    current_platform = spack.platforms.host()
+    if not spec.satisfies(f"platform={current_platform}"):
```

Comment:
`satisfies` is a bit too much for this, plus it's a temporary spec you're creating 1000s of times for large caches.

Reply:
True, this needs to be optimized, and also the target part. I plan to do that and take care of the monkey patching issues in unit tests 🙂

Reply:
I'll keep this for another PR.